app/models/library.rb

```diff
@@ -6,4 +6,8 @@ class Library < ApplicationRecord
   validates :name, presence: true
   validates :id,  numericality: { only_integer: true, allow_nil: true }
   validates :description, length: { maximum: 500, message: "%{value} is too long, please shorten the description to 500 characters." }, allow_blank: true
+
+  def opt_groups
```

Comment:
Is this still being used?  I think it was replaced by the method in the user_permissions controller.